gap/io.gi

```diff
@@ -65,7 +65,11 @@ InstallMethod( ViewObj, "for an IO_Result",
 IO.OpenFiles := Set([]);
 InstallAtExit(function()
   local file;
-  for file in IO.OpenFiles do
+  # Calling IO_Close below removes the file from IO.OpenFiles,
+  # so iterating over IO.OpenFiles directly would skip every
+  # second file. To avoid this, we iterate over a shallow copy
+  # of IO.OpenFiles.
+  for file in ShallowCopy( IO.OpenFiles ) do
```

Comment:
Good catch!

Alternative implementation:
```suggestion
  while Length(IO.OpenFiles) > 0 do
      file := Remove(IO.OpenFiles);
```
This also avoids lots of unnecessary memory shuffling (though if you have so many files open that this would matter, you probably have other problems)...

Reply:
I think modifying `IO.OpenFiles` should only be done by the `IO_...` functions to ensure consistency, so I do not want to modify it by calling `Remove(IO.OpenFiles);`. I have pushed a different suggestion which avoids `ShallowCopy`, too.